library/Core/src/Juvix/Core/Erasure/Algorithm.hs

```diff
@@ -9,24 +9,28 @@ import Juvix.Library hiding (empty)
 import qualified Juvix.Library.NameSymbol as NameSymbol
 import qualified Juvix.Library.Usage as Usage
 
-type ErasureM primTy primVal m =
+type ErasureM primTy1 primTy2 primVal1 primVal2 m =
```

Comment:
I'm guesing we take 4 aprams as we have to track before and after types?

Reply:
I guess? Why are before and after types different (now)?

Reply:
The base primitives themselves (Michelson instructions, or whatever) don't change, but the variable type in partial applications does, so `primVal` goes from e.g. `Return' IR.NoExt primTy primVal` (where arguments can contain dB indices) to `Return' ErasedAnn.T primTy primVal` (where they can't).